packages/-ember-data/tests/integration/references/record-test.js

```diff
@@ -32,75 +33,134 @@ module('integration/references/record', function (hooks) {
     assert.equal(recordReference.id(), 1);
   });
 
-  test('push(object)', function (assert) {
-    var done = assert.async();
+  test('a RecordReference can be retrieved via store.getReference(identifier) without local state', function (assert) {
+    let store = this.owner.lookup('service:store');
+    let recordReference = store.getReference({ type: 'person', id: '1' });
+
+    assert.equal(recordReference.remoteType(), 'identity');
+    assert.equal(recordReference.type, 'person');
+    assert.equal(recordReference.id(), '1');
+  });
+
+  [
+    { withType: true, withId: true, withLid: true, desc: 'type, id and lid' },
+    { withType: true, withLid: true, desc: 'type and lid' },
+    { withType: true, withLid: true, isCreate: true, desc: 'type and lid via store.createRecord (no local id)' },
+    {
+      withType: true,
+      withLid: true,
+      isCreate: true,
+      generateLid: true,
+      desc: 'type and generated lid via store.createRecord (no local id)',
+    },
+    { withType: true, withLid: true, generateLid: true, desc: 'type and generated lid' },
+    { withType: true, withId: true, withLid: true, desc: 'type, id and lid from cache' },
+    { withType: true, withLid: true, exta: { id: null }, desc: 'type, null id, and lid' },
+    {
+      withType: true,
+      withLid: true,
+      exta: { id: null },
+      isCreate: true,
+      desc: 'type, null id, and lid via store.createRecord',
+    },
+  ].forEach(({ withType, withId, withLid, extra, isCreate, generateLid, desc }) => {
+    test(`a RecordReference can be retrieved with ${desc}`, function (assert) {
+      let store = this.owner.lookup('service:store');
+      let person;
+      if (isCreate) {
+        // no id
+        person = store.createRecord('person');
+      } else {
+        person = store.push({
+          data: {
+            type: 'person',
+            id: '1',
+            attributes: {
+              name: 'le name',
+            },
+          },
+        });
+      }
+
+      const getReferenceArgs = Object.create(null);
+      if (withType) {
+        getReferenceArgs.type = 'person';
+      }
+      if (withId && !isCreate) {
+        getReferenceArgs.id = '1';
+      }
+      if (withLid) {
+        if (generateLid) {
```

Comment:
I think this can be simplified to `!isCreate`, and I think we can remove one of the test scenarios from the array config above. we'll always generate in the non-create case. I don't think we need to worry about testing the output of the cache generation vs `recordIdentifierFor` as other tests cover and ensure that these things match.